rust/src/treefile.rs

```diff
@@ -1226,6 +1259,66 @@ etc-group-members:
                 == "table"
         );
     }
+
+    #[test]
+    fn test_split_whitespace_unless_quoted() {
+        // test single quoted package
+        let single_quoted_pkg = String::from("'foobar >= 1.0'");
+        let pkgs: Vec<String> = split_whitespace_unless_quoted(&single_quoted_pkg).collect();
+        assert_eq!("foobar >= 1.0", pkgs[0]);
+
+        // test multiple quoted packages
+        let mult_quoted_pkg = String::from("'foobar >= 1.0' 'quuz < 0.5' 'corge > 2'");
+        let pkgs: Vec<String> = split_whitespace_unless_quoted(&mult_quoted_pkg).collect();
```

Comment:
Then once it returns `Result<>` this call would be:
`let pkgs: Vec<String> = split_whitespace_unless_quoted(&mult_quoted_pkg)?.collect();`
(Notice the added `?`)